File: client/windows/network/websocket_client.cpp
#include "websocket_client.h"
#include <iostream>

bool WebSocketClient::connect(const std::string& url)
{
    std::cout << "connect " << url << std::endl;
    return true;
}

void WebSocketClient::login(const std::string& id,const std::string& name)
{
    std::cout << "login " << id << " " << name << std::endl;
}

File: client/src/network/WebSocketClient.js
class WebSocketClient {
  constructor(url) {
    this.url = url;
    this.socket = null;
    this.handlers = {};
  }

  connect() {
    this.socket = new WebSocket(this.url);

    this.socket.onmessage = (event) => {
      const message = JSON.parse(event.data);
      const handler = this.handlers[message.type];
      if (handler) {
        handler(message);
      }
    };
  }

  send(type, data = {}) {
    if (!this.socket) return;

    this.socket.send(JSON.stringify({
      type,
      data,
      timestamp: Date.now()
    }));
  }

  on(type, callback) {
    this.handlers[type] = callback;
  }
}

module.exports = WebSocketClient;
